crates/core/src/kernel/mod.rs

```diff
@@ -13,6 +13,7 @@ pub mod schema;
 mod snapshot;
 pub mod transaction;
 
+pub use delta_kernel::engine;
```

Comment:
do we need to expose `engine` to the outside world? Ultimately my hope wpuld be that we can hide the fact that an engine exists from end users. Particularly now, since there are still some footguns with "hidden async" when using engine ...